app/src/main/java/com/pose/move/network/ApiOperationWrapper.kt

```diff
@@ -0,0 +1,42 @@
+package com.pose.move.network
+
+import android.util.Log
+import com.squareup.moshi.Moshi
+import java.io.IOException
+import javax.inject.Inject
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.withContext
+import retrofit2.HttpException
+import kotlin.coroutines.CoroutineContext
+
+class ApiOperationWrapper @Inject constructor(moshi: Moshi) {
+
+    val errorJsonAdapter: ErrorResponseJsonAdapter = ErrorResponseJsonAdapter(moshi)
+
+    suspend inline operator fun <T> invoke(
+        coroutineContext: CoroutineContext = IO,
```

Comment:
Hide the kotlin's Coroutine Dispatchers behind your own implementation.

You could have an interface like
```
interface MoveDispacters {
    val io: CoroutineContext
    val default: CoroutineContext
    val main: CoroutineContext
}
```

And inject it everywhere. This way, you don't depend directly on the platform, but rather abstract your whole implementation. This also means that if you ever want to change a dispatcher, it'll be very easy to do so.

Reply:
I remembered that you could also use Annotations to tell Hilt which Dispatchers to inject. 
Either that or use the above solution